packages/core/react/src/WalletProvider.tsx

```diff
@@ -78,7 +78,9 @@ export function WalletProvider({
     }, [adaptersWithStandardAdapters, mobileWalletAdapter]);
     const [walletName, setWalletName] = useLocalStorage<WalletName | null>(
         localStorageKey,
-        getIsMobile(adaptersWithStandardAdapters) ? SolanaMobileWalletAdapterWalletName : null
+        adaptersWithStandardAdapters.length === 0 && getIsMobile(adaptersWithStandardAdapters)
```

Comment:
Hmm. This has the effect of, if _any adapter_ is included, to not select MWA. The problem with this in practice is that almost all dapps will include adapters, even ones they don't need because they've been deprecated by Standard Wallets, or ones that simply don't work at all on the platform. This is especially true because most dapps I've seen don't use separate mobile builds with conditional compilation of included adapters.

This basically means that it's always going to be `false`, and we're never even going to do the `getIsMobile` check. Now, I'm probably fine with eliminating all preference for MWA, but I want to be clear that's what we're doing here, and if so, we might as well just do it outright and remove these checks.